.github/workflows/web_staging.yml

```
@@ -0,0 +1,207 @@
+          echo "AWS_SECRET_ACCESS_KEY=${{ secrets.AWS_SECRET_ACCESS_KEY }}" >> $GITHUB_ENV
+          echo "AWS_CLOUDFRONT_DISTRIBUTION_ID=${{ secrets.AWS_CLOUDFRONT_DISTRIBUTION_ID }}" >> $GITHUB_ENV
+          echo "AWS_DEFAULT_REGION=${{ secrets.AWS_DEFAULT_REGION }}" >> $GITHUB_ENV
+          echo "AWS_S3_BUCKET=${{ secrets.AWS_S3_BUCKET }}" >> $GITHUB_ENV
+
+      - name: "[deployment] Install dependencies"
+        run: |
+          pushd /tmp >/dev/null
+            curl -fsSL "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+            unzip -oqq awscliv2.zip
+            sudo ./aws/install --update
+          popd >/dev/null
+          aws --version
+
+      - name: "[deployment] Clear AWS S3 bucket: html files, web root, app bundle, keep checksums"
+        run: |
+          aws s3 rm s3://${AWS_S3_BUCKET} --recursive --exclude "proteins/*" --exclude "acknowledgements/*" --exclude "checksums-ack-s3.txt"
+
+      - name: "[deployment] Copy to AWS S3: app bundle"
+        run: |
+          cd web/.build/production/web
+          aws s3 cp --recursive --cache-control "max-age=2592000, public" "_next/" "s3://${AWS_S3_BUCKET}/_next/"
+
+      - name: "[deployment] Copy to AWS S3: web root"
+        run: |
+          cd web/.build/production/web
+          aws s3 cp --recursive \
+            --exclude "_next/*" \
+            --exclude "*.html" \
+            --exclude "acknowledgements/*" \
+            --exclude "proteins/*" \
+          "./" "s3://${AWS_S3_BUCKET}/"
+
+      - name: "[deployment] Copy to AWS S3: html files"
+        run: |
+          cd web/.build/production/web
+          find * -type f -name "*.html" -print0 | xargs -0 -P4 -n1 -I '{}' -- bash -c '\
+            file={}; \
+            aws s3 cp \
+              --content-type "text/html" \
+              --cache-control "no-cache" \
+              --metadata-directive REPLACE \
+                $file \
+                s3://${AWS_S3_BUCKET}/${file%.html}'
+
+      - name: "[deployment] Invalidate AWS Cloudfront cache: html files, web root, app bundle"
+        run: |
+          aws cloudfront create-invalidation \
+            --distribution-id ${AWS_CLOUDFRONT_DISTRIBUTION_ID} \
+            --paths "/*"
+
+      - name: "[deployment] Compare acknowledgements checksums"
+        run: |
+          cd web
+          if aws s3 ls s3://${AWS_S3_BUCKET}/checksums-ack-s3.txt; then
+            echo "s3 checksums found, downloading s3 checksums"
+            aws s3api get-object --bucket ${AWS_S3_BUCKET} --key checksums-ack-s3.txt checksums-ack-s3.txt
+          else
+            echo "no s3 checksums found, using ci checksums"
+            touch checksums-ack-s3.txt
+          fi
+          echo "comparing checksums"
+          comm -23 checksums-ack-ci.txt checksums-ack-s3.txt | awk '{print $1}' > outdated_acknowledgements.txt
+          echo "found outdated files"
```

This is maybe a bit confusing, since it will also say "found outdated files" if it didn't find any.
```suggestion
          echo "Listing outdated files:"
```